index.bs

```diff
@@ -424,58 +433,142 @@ When this method is invoked, the user agent MUST execute the following algorithm
-    - If `current.type` does not contain a {{ScopedCredentialType}} supported by this implementation, then stop processing
-        |current| and move on to the next element in {{cryptoParameters}}.
-    - Let |normalizedAlgorithm| be the result of <a>normalizing an algorithm</a> [[!WebCryptoAPI]],
-        with |alg| set to `current.algorithm` and |op| set to 'generateKey'. If an error occurs during this
-        procedure, then stop processing |current| and move on to the next element in {{cryptoParameters}}.
-    - Add a new object of type {{ScopedCredentialParameters}} to |normalizedParameters|, with |type| set to `current.type` and
-        |algorithm| set to |normalizedAlgorithm|.
-
-1. If |normalizedAlgorithm| is empty and {{cryptoParameters}} was not empty, cancel the timer started in step 2, reject
-    |promise| with a DOMException whose name is "{{NotSupportedError}}", and terminate this algorithm.
-
-1. If the {{ScopedCredentialOptions/extensions}} member of {{options}} is <a>present</a>, process any extensions supported by
-    this client platform, to produce the extension data that needs to be sent to the authenticator. If an error is encountered
-    while processing an extension, skip that extension and do not produce any extension data for it. Call the result of this
-    processing |clientExtensions|.
+    Issue(w3c/webauthn#259): The rest of this algorithm assumes |rpId| is an
+    [=origin=], but the above step sometimes produces a string.
+
+1. Let |normalizedParameters| be a new [=list=] whose [=list/items=] are pairs of
+    ScopedCredentialType and a [=dictionary=] type (as returned by [=normalizing
+    an algorithm=]).
+1. [=list/For each=] |current| of {{cryptoParameters}}:
+    1. If <code>|current|.{{ScopedCredentialParameters/type}}</code> does not
+        contain a {{ScopedCredentialType}} supported by this implementation,
+        then [=continue=].
+    1. Let |normalizedAlgorithm| be the result of <a>normalizing an algorithm</a>
+        [[!WebCryptoAPI]], with |alg| set to
+        <code>|current|.{{algorithm}}</code> and |op| set to `"generateKey"`. If
+        an error occurs during this procedure, then [=continue=].
+    1. [=list/Append=] the pair of
+        <code>|current|.{{ScopedCredentialParameters/type}}</code> and
+        |algorithm| to |normalizedAlgorithm|.
+
+1. If |normalizedParameters| is empty and {{cryptoParameters}} was not empty,
+    cancel the timer started in step 2, return [=a promise rejected with=] with
+    a {{DOMException}} whose name is "{{NotSupportedError}}", and terminate this
+    algorithm.
+
+1. Let |clientExtensions| be a new [=list=].
+1. If the {{ScopedCredentialOptions/extensions}} member of {{options}} is
+    <a>present</a>, then [=map/for each=] |extension| → |argument| of
+    <code>{{options}}.{{ScopedCredentialOptions/extensions}}</code>:
+    1. If |extension| is not supported by this client platform, [=continue=].
```

Comment:
given https://github.com/w3c/webauthn/pull/347#discussion_r103022672, should this substep be something like..

If |extension| is not supported by this client platform, either [=continue=], or optionally CBOR-encode |argument| as a byte string, and go to Step 4.

..@vijaybh ?